oxd-server/src/test/java/org/gluu/oxd/server/service/RpSyncServiceTest.java

```diff
@@ -0,0 +1,117 @@
+import java.util.Date;
+
+@Guice(modules = GuiceModule.class)
+public class RpSyncServiceTest {
+
+    @Inject
+    ConfigurationService configurationService;
+    @Inject
+    RpService service;
+    @Inject
+    RpSyncService rpSyncService;
+    @Inject
+    PersistenceService persistenceService;
+
+    @BeforeClass
+    public void setUp() throws IOException, ConfigurationException {
+        configurationService.setConfiguration(parseConfiguration(ResourceHelpers.resourceFilePath("oxd-server-jenkins.yml")));
+        persistenceService.create();
+        service.removeAllRps();
+        service.load();
+    }
+
+    @AfterSuite
+    public void tearDownSuite() {
+        persistenceService.destroy();
+    }
+
+    @Test
+    public void testShouldSyncForNullRp() {
+        Assert.assertFalse(rpSyncService.shouldSync(null));
+        service.removeAllRps();
+    }
+
+    @Test
+    public void testShouldSyncForNotNullRp() {
+        Rp rp = new Rp();
+        rp.setSyncClientFromOp(true);
+        //adding date 10 mintutes behind current time and setting as LastSynced
+        rp.setLastSynced(Utils.addTimeToDate(new Date(), -10, Calendar.MINUTE));
+        rp.setSyncClientPeriodInSeconds(5 * 60);
+        Assert.assertTrue(rpSyncService.shouldSync(rp));
+    }
+
+    @Test
+    public void testShouldSyncForSyncAllowed() {
+        Rp rp = new Rp();
+        rp.setSyncClientFromOp(true);
+        //adding date 10 mintutes behind current time and setting as LastSynced
+        rp.setLastSynced(Utils.addTimeToDate(new Date(), -10, Calendar.MINUTE));
+        rp.setSyncClientPeriodInSeconds(5 * 60);
+        Assert.assertTrue(rpSyncService.shouldSync(rp));
+    }
+
+    @Test
+    public void testShouldSyncForSyncNotAllowed() {
+        Rp rp = new Rp();
+        rp.setSyncClientFromOp(false);
+        //adding date 10 mintutes behind current time and setting as LastSynced
+        rp.setLastSynced(Utils.addTimeToDate(new Date(), -10, Calendar.MINUTE));
+        rp.setSyncClientPeriodInSeconds(5 * 60);
+        Assert.assertFalse(rpSyncService.shouldSync(rp));
+    }
+
+    @Test(expectedExceptions = {NullPointerException.class})
+    public void testShouldSyncForNullLastSynced() {
+        Rp rp = new Rp();
+        rp.setSyncClientFromOp(true);
+        rp.setLastSynced(null);
+        rp.setSyncClientPeriodInSeconds(5 * 60);
+        Assert.assertTrue(rpSyncService.shouldSync(rp));
+    }
+
+    @Test
+    public void testShouldSyncForNotNullLastSynced() {
+        Rp rp = new Rp();
+        rp.setSyncClientFromOp(true);
+        //adding date 10 mintutes behind current time and setting as LastSynced
+        rp.setLastSynced(Utils.addTimeToDate(new Date(), -10, Calendar.MINUTE));
+        rp.setSyncClientPeriodInSeconds(5 * 60);
+        Assert.assertTrue(rpSyncService.shouldSync(rp));
+    }
+
+    private static OxdServerConfiguration parseConfiguration(String pathToYaml) throws IOException, ConfigurationException {
```

Comment:
It's copy/paste. Please re-use code. Create separate class and remove all duplicates.

Reply:
Done